include/aws/common/array_list.h

```diff
@@ -59,6 +59,19 @@ void aws_array_list_init_static(
     size_t item_count,
     size_t item_size);
 
+/**
+ * Initializes an array list with a preallocated array of *already-initialized* elements. item_count is the number of
+ * elements in the array, and item_size is the size in bytes of each element.
+ *
+ * Primary use is to be able to treat an already initialized C array as an array list.
```

Comment:
seriously: would it make more sense to just use the C array as it is? what benefit do you get by forcing people to use the array-list API to access the data? It checks your bounds, but we seldom write code anymore that checks whether of the functions returned an error

Reply:
It's not my choice. The mqtt311 API throws around array lists in places and it's useful in tests.